torchvision/models/vision_transformer.py

```diff
@@ -148,11 +158,32 @@ def __init__(
         self.representation_size = representation_size
         self.norm_layer = norm_layer
 
-        input_channels = 3
-
-        # The conv_proj is a more efficient version of reshaping, permuting
-        # and projecting the input
-        self.conv_proj = nn.Conv2d(input_channels, hidden_dim, kernel_size=patch_size, stride=patch_size)
+        if conv_stem_layers is not None:
+            # As per https://arxiv.org/abs/2106.14881
+            self.conv_proj = nn.Sequential()
+            prev_channels = 3
+            for i, conv_stem_layer in enumerate(conv_stem_layers):
+                self.conv_proj.add_module(
+                    f"conv_bn_relu_{i}",
+                    ConvNormActivation(
+                        in_channels=prev_channels,
+                        out_channels=conv_stem_layer.out_channels,
+                        kernel_size=conv_stem_layer.kernel_size,
+                        stride=conv_stem_layer.stride,
+                        padding=conv_stem_layer.padding,
+                        norm_layer=nn.BatchNorm2d,
+                        activation_layer=nn.ReLU,
+                        bias=conv_stem_layer.bias,
+                    ),
+                )
+                prev_channels = conv_stem_layer.out_channels
+            self.conv_proj.add_module(
+                "conv_last", nn.Conv2d(in_channels=prev_channels, out_channels=hidden_dim, kernel_size=1)
+            )
+        else:
+            self.conv_proj = nn.Conv2d(  # type: ignore
```

Comment:
Ignoring is usually a last ditch attempt. We should fix properly.

Try the following: Define use a local variable `conv_proj` inside the if/else statement. Then right after the end of the if do:
`self.conv_proj: nn.Module = conv_proj`